cmd/web-handlers.go

```diff
@@ -569,17 +569,22 @@ func (web *webAPIHandlers) DownloadZip(w http.ResponseWriter, r *http.Request) {
 
 	token := r.URL.Query().Get("token")
 
-	if !isAuthTokenValid(token) {
-		writeWebErrorResponse(w, errAuthentication)
-		return
-	}
 	var args DownloadZipArgs
 	decodeErr := json.NewDecoder(r.Body).Decode(&args)
 	if decodeErr != nil {
 		writeWebErrorResponse(w, decodeErr)
 		return
 	}
 
```

Comment:
We might have to add some comment here on why we are validating the token later i.e we are reading from the body that the client sent and then failing. Perhaps we can also do a `io.LimitReader(r.Body)` on the maximum allowed.. body header for this one. So that the server doesn't crash due to some erroneous handcrafted client which can happen now. Since token is validated later..